src/RpcClient/Nep17API.cs

```diff
@@ -141,9 +146,15 @@ public async Task<Transaction> CreateTransferTxAsync(UInt160 scriptHash, KeyPair
 
             TransactionManagerFactory factory = new TransactionManagerFactory(rpcClient);
             TransactionManager manager = await factory.MakeTransactionAsync(script, signers).ConfigureAwait(false);
-            return await manager
-                .AddSignature(fromKey)
-                .SignAsync().ConfigureAwait(false);
+
+            var vmResult = await rpcClient.InvokeScriptAsync(script, signers).ConfigureAwait(false);
+
+            if (ErrorOnTransferFalse == false || (vmResult.State == VMState.HALT && vmResult.Stack?[0].GetBoolean() == true))
```

Comment:
I think usually this is solved by adding an `ASSERT` to the end of the script (which is trivial here since the script is created in the same function`).

Reply:
It's better to do it like @roman-khimov said